### DISCUSSION TITLE
I want to know where a connection has been closed on the client side

### DISCUSSION THREAD

**Discussion by bnjjj** [tokio-rs/axum#1726] [Q&A]
Hi I'm currently using stream to return data to the client with Axum. Here is a snippet of my code:

```rust
(parts, StreamBody::new(body)).into_response().map(|body| {
    // Axum makes this `body` have type:
    // https://docs.rs/http-body/0.4.5/http_body/combinators/struct.UnsyncBoxBody.html
    let mut body = Box::pin(body);
    // We make a stream based on its `poll_data` method
    // in order to create a `hyper::Body`.
    Body::wrap_stream(stream::poll_fn(move |ctx| {
        body.as_mut().poll_data(ctx)
    }))
})
```

I don't know how can I do to catch a closed connection from the client ? Do you have an idea ?

Thanks a lot

**Comment by davidpdrsn** [comment]
Why do you need to know? The body will automatically be dropped and not pulled anymore if the client disconnects.

**Reply by bnjjj** [reply]
I spawn a task listening on a channel and I want to close it if the connection is closed. Is there a callback on a stream which is triggered when it's dropped ?

**Reply by davidpdrsn** [reply]
The channel's send/receive call will fail if the other half has been dropped. So should happen by itself.

**Reply by deleted** [reply]
@davidpdrsn 

I was going nuts trying to figure out why the tokio task i was using to stream response data from the DB (via a MPSC channel) would terminate on a client disconnect when an arbirtrary print-and-sleep loop task launched by the Axum handler didn't, and this comment enlightened me. 

So thank you!
